airflow/www/views.py

```diff
@@ -110,6 +110,15 @@
 FILTER_STATUS_COOKIE = 'dag_status_filter'
 
 
+def get_truncate_task_duration(task_duration):
+    """
+    Cast task_duration to an int was for optimization for large/huge dags if task_duration > 10s
+    otherwise we keep it as a float with 3dp(truncate)
+    """
+    duration = int(task_duration) if task_duration > 10.0 else math.trunc(task_duration * 1000) / 1000.0
+    return duration
```

Comment:
```suggestion
def truncate_task_duration(task_duration):
    """
    Cast task_duration to an int was for optimization for large/huge dags if task_duration > 10s
    otherwise we keep it as a float with 3dp
    """
    return int(task_duration) if task_duration > 10.0 else round(task_duration, 3)
```

Reply:
I didn't use `round` just because the 
truncate_task_duration(9.9996) = 10.0
truncate_task_duration(10.1231) = 10

I thought the truncate looks more straightforward when comparing to the database record.

Reply:
I think for the one edge case of (`9.999` getting rounded to `10.0`) I'd rather easier to read code.